diff --git a/src/makefile.all b/src/makefile.all
index d158664..f840962 100644
--- a/src/makefile.all
+++ b/src/makefile.all
@@ -252,7 +252,7 @@ PKT_STUB = pkt_stub.h
 prefix = /dev/env/DJDIR/net/watt
 
 CFLAGS = -O3 -g -I. -I../inc -DWATT32_BUILD -W -Wall -Wno-strict-aliasing \
-         -march=i386 -mtune=i586
+         -march=i386 -mtune=i586 $(CFLAGS_FOR_TARGET)
 
 STAT_LIB = ../lib/libwatt.a
 OBJDIR   = build/djgpp
